src/Umbraco.Core/Strategies/RelateOnTrashHandler.cs

```diff
@@ -72,5 +89,46 @@ private void ContentService_Trashed(IContentService sender, MoveEventArgs<IConte
                 
             }
         }
+
+        private void MediaService_Trashed(IMediaService sender, MoveEventArgs<IMedia> e)
+        {
+            var relationService = ApplicationContext.Current.Services.RelationService;
+            var entityService = ApplicationContext.Current.Services.EntityService;
+            var relationTypeAlias = Constants.Conventions.RelationTypes.RelateParentMediaFolderOnDeleteAlias;
+            var relationType = relationService.GetRelationTypeByAlias(relationTypeAlias);
+
+            // check that the relation-type exists, if not, then recreate it
+            if (relationType == null)
+            {
+                var documentObjectType = new Guid(Constants.ObjectTypes.Document);
+                var relationTypeName = Constants.Conventions.RelationTypes.RelateParentMediaFolderOnDeleteName;
+
+                relationType = new RelationType(documentObjectType, documentObjectType, relationTypeAlias, relationTypeName);
+                relationService.Save(relationType);
+            }
+
+            foreach (var item in e.MoveInfoCollection)
+            {
+                var originalPath = item.OriginalPath.ToDelimitedList();
+                var originalParentId = originalPath.Count > 2
+                    ? int.Parse(originalPath[originalPath.Count - 2])
+                    : Constants.System.Root;
+
+                //before we can create this relation, we need to ensure that the original parent still exists which
+                //may not be the case if the encompassing transaction also deleted it when this item was moved to the bin
+
+                if (entityService.Exists(originalParentId))
+                {
+                    // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                    var relation = new Relation(originalParentId, item.Entity.Id, relationType);
+                    relationService.Save(relation);
+
+                    ApplicationContext.Current.Services.AuditService.Add(AuditType.Delete,
+                        string.Format("Trashed media item with Id: '{0}' related to original parent media item with Id: '{1}'", item.Entity.Id, originalParentId),
```

Comment:
Can you move this text to the language files and use the localization service to get it?